src/downzip-sw.js

```diff
@@ -77,13 +77,29 @@ self.addEventListener('fetch', async (event) => {
         for(let i=0; i<zipMap[id].files.length; i++){
             const file = zipMap[id].files[i]
 
-            // Start new file in the zip
-            zipMap[id].zip.startFile(file.name)
-
+            let name = ''
             // Append all the downloaded data
             try {
                 await new Promise((resolve, reject) => {
-                    fetch(file.downloadUrl).then(response => response.body).then(async (stream) => {
+                    fetch(file.downloadUrl).then(async response => {
+                        if (!response.ok) {
+                            Utils.error(`downloadUrl: ${file.downloadUrl}, response status: ${response.status}`)
+                            return resolve()
+                        } 
+
+                        name = file.name
+                        const extension = name.match(/\.(\w+)(\?|$|%3F)/)
+                        if (!extension) {
+                            const newExtension = response.headers.get('Content-Type').split('/').pop()
+                            name = file.name + '.' + newExtension
+                        }
+
+                        // Start new file in the zip
+                        zipMap[id].zip.startFile(name)

```

Comment:
@GProst Handling no file extension here. Had to move the startFile and endFile methods into the fetch promise.